cmd/snap-repair/runner.go

```diff
@@ -43,6 +46,91 @@ import (
 	"github.com/snapcore/snapd/strutil"
 )
 
+// Repair is a runnable repair.
+type Repair struct {
+	*asserts.Repair
+
+	run      *Runner
+	sequence int
+}
+
+// SetStatus sets the status of the repair in the state and saves the latter.
+func (r *Repair) SetStatus(status RepairStatus) {
+	brandID := r.BrandID()
+	cur := *r.run.state.Sequences[brandID][r.sequence-1]
+	cur.Status = status
+	r.run.setRepairState(brandID, cur)
+	r.run.SaveState()
+}
+
+// Run executes the repair script leaving execution trail files on disk.
+func (r *Repair) Run() error {
+	// write the script to disk
+	rundir := filepath.Join(dirs.SnapRepairRunDir, r.BrandID(), r.RepairID())
+	err := os.MkdirAll(rundir, 0775)
+	if err != nil {
+		return err
+	}
+
+	script := filepath.Join(rundir, fmt.Sprintf("script.r%d", r.Revision()))
+	err = osutil.AtomicWriteFile(script, r.Body(), 0700, 0)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now().Unix()
+	logPath := filepath.Join(rundir, fmt.Sprintf("r%d.%v.output", r.Revision(), now))
```

Comment:
as I mentioned on Friday maybe we should also have an incremental counter (000001 etc) in these names, because we cannot always trust now,  also wondering if we should put now in a more readable format?  2017-08-21T224455 ?